src/auth/oauth2/qgso2.cpp

```diff
@@ -69,7 +69,8 @@ void QgsO2::initOAuthConfig()
   mIsLocalHost = isLocalHost( QUrl( localpolicy.arg( mOAuth2Config->redirectPort() ) ) );
 
   setTokenUrl( mOAuth2Config->tokenUrl() );
-  setRefreshTokenUrl( mOAuth2Config->refreshTokenUrl() );
+  // refresh token url is marked as optional -- we use the token url if user has not specified a specific refresh URL
```

Comment:
This doesn't look right to me: refresh URL is optional because the refresh operation is not always supported, but I'm pretty sure that if the refresh endpoint exists, it cannot be the same URL of the token URL.